src/p09.rs

```diff
@@ -0,0 +1,87 @@
+use std::{
+    collections::{HashSet, VecDeque},
+    error::Error,
+};
+
+fn get_neighbor_points(map: &[String], pos: (usize, usize)) -> Vec<(usize, usize)> {
+    let mut neighbors = Vec::new();
+
+    if pos.0 > 0 {
+        neighbors.push((pos.0 - 1, pos.1));
+    }
+    if pos.0 < map.len() - 1 {
+        neighbors.push((pos.0 + 1, pos.1));
+    }
+    if pos.1 > 0 {
+        neighbors.push((pos.0, pos.1 - 1));
+    }
+    if pos.1 < map[0].len() - 1 {
+        neighbors.push((pos.0, pos.1 + 1));
+    }
+
+    neighbors
+}
+
+fn get_low_points(map: &[String]) -> Vec<(usize, usize)> {
+    let mut low_points = Vec::new();
+
+    for y in 0..map.len() {
+        for x in 0..map[0].len() {
+            if get_neighbor_points(map, (y, x))
+                .iter()
+                .all(|p| map[p.0].chars().nth(p.1).unwrap() > map[y].chars().nth(x).unwrap())
+            {
+                low_points.push((y, x));
+            }
+        }
+    }
+
+    low_points
+}
+
+fn get_basin_size(map: &[String], start: (usize, usize)) -> i64 {
```

Comment:
```suggestion
fn get_basin_size(map: &[&str], start: (usize, usize)) -> i64 {
```
we usually do this and call using .as_deref() on the Option at the call site

Reply:
There is no `Option` at the call site :thinking: